### PR TITLE
user roles list filter and refactor

### PR DESCRIPTION
**Pull request by xnetsrak**
## Description

user roles list filter and refactor
## References

- #0000

### CODE REVIEW

**Comment by mimse** [comment]
image
Getting this error

File: libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab-list-filter.component.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  ).map((key) => ({
+    displayValue: key,
+    value: key,
+  }));
+
+  constructor() {
+    this.init();
+  }
+
+  init = () => {
```
Comment (mimse) [inline libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab-list-filter.component.ts:83]: Suggestion: maybe use OnInit interface?

File: libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.scss [hunk 2]
```diff
@@ -16,10 +16,21 @@
  */
 :host {
   display: block;
-  // TODO: Add spacing variable for 24px
   margin: 24px var(--watt-space-s);
 }
 
+.user-roles {
+  &__spinner {
```
Comment (mimse) [inline libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.scss:23]: Question: Haven't seen the conversion with __ in class names. Do we use this elsewhere?

File: libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.ts [hunk 3]
```diff
@@ -14,30 +14,65 @@
...
 export class DhUserRolesTabComponent {
-  @Input() roles: UserRoleInfoDto[] = [];
+  private readonly store = inject(DhAdminUserRolesManagementDataAccessApiStore);
+  roles$ = this.store.rolesFiltered$;
+
+  isLoading$ = this.store.isLoading$;
+  hasGeneralError$ = this.store.hasGeneralError$;
+
+  updateFilterStatus(status: string | null) {
+    const statusParsed = status as UserRoleStatus;
```
Comment (mimse) [inline libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.ts:66]: Question: Are they save casts?

**Review by mimse — COMMENTED** [review]
Looks good, have some suggestions and questions :-)

File: libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.scss (hunk 2 above)
Comment (xnetsrak) [reply libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.scss:23]: yeah two other places, it is copy paste

File: libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.ts (hunk 3 above)
Comment (xnetsrak) [reply libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.ts:66]: what do you mean ? yes it works with values and null.

File: libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.scss (hunk 2 above)
Comment (dzhavat) [reply libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.scss:23]: For context, this class name is following [BEM](https://css-tricks.com/bem-101/) methodology. We are not really consistent with BEM, but it's being used in quite a few places.

File: libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.ts (hunk 3 above)
Comment (dzhavat) [reply libs/dh/admin/feature-user-management/src/lib/tabs/roles-tab/dh-roles-tab.component.ts:66]: An issue here is that we can get `null` from the dropdown that is also cast to `UserRoleStatus`. Maybe we should send the values as-is and have another way for handling `null` values in the store.

**Review by dzhavat — APPROVED** [review]

**Review by mimse — APPROVED** [review]

**Pull request merged**